### PR TITLE
Update docker support for k8s

### PR DESCRIPTION
**Pull request by xichengliudui**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
https://github.com/kubernetes/kubernetes/pull/72823

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xichengliudui** [comment]
I found that @dims might be missing these two files, I modified them

File: cmd/kubeadm/app/util/system/types_unix.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ var DefaultSysSpec = SysSpec{
 	Cgroups: []string{"cpu", "cpuacct", "cpuset", "devices", "freezer", "memory"},
 	RuntimeSpec: RuntimeSpec{
 		DockerSpec: &DockerSpec{
-			Version:     []string{`1\.1[1-3]\..*`, `17\.0[3,6,9]\..*`, `18\.06\..*`},
+			Version:     []string{`1\.1[1-3]\..*`, `17\.0[3,6,9]\..*`, `18\.09\..*`}, //Requires [18.09] or later
```
Comment (neolit123) [inline cmd/kubeadm/app/util/system/types_unix.go]: it has to be:
```
`18\.0[6,9]\..*`
```

File: cmd/kubeadm/app/util/system/types_windows.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ var DefaultSysSpec = SysSpec{
 	Cgroups: []string{},
 	RuntimeSpec: RuntimeSpec{
 		DockerSpec: &DockerSpec{
-			Version:     []string{`18\.06\..*`}, //Requires [18.06] or later
+			Version:     []string{`18\.09\..*`}, //Requires [18.09] or later
```
Comment (neolit123) [inline cmd/kubeadm/app/util/system/types_windows.go]: let's make this one:
`18\.0[6,9]\..*`

you also need to modify:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/system/docker_validator_test.go#L31

and make -> `18\.06\..*` -> `18\.0[6,9]\..*`

**Review by neolit123 — COMMENTED** [review]
i saw this problem yesterday @xichengliudui 
thank you for the PR. please see my comments.

**Comment by xichengliudui** [comment]
@neolit123  Thank you very much for your advice!

File: cmd/kubeadm/app/util/system/types_unix.go [hunk 3]
```diff
@@ -62,7 +62,7 @@ var DefaultSysSpec = SysSpec{
 	Cgroups: []string{"cpu", "cpuacct", "cpuset", "devices", "freezer", "memory"},
 	RuntimeSpec: RuntimeSpec{
 		DockerSpec: &DockerSpec{
-			Version:     []string{`1\.1[1-3]\..*`, `17\.0[3,6,9]\..*`, `18\.06\..*`},
+			Version:     []string{`1\.1[1-3]\..*`, `17\.0[3,6,9]\..*`, `18\.0[6,9]\..*`}, //Requires [18.09] or later
```
Comment (neolit123) [inline cmd/kubeadm/app/util/system/types_unix.go]: please remove the `Requires [18.09] or later` comment.

File: cmd/kubeadm/app/util/system/types_windows.go [hunk 4]
```diff
@@ -38,7 +38,7 @@ var DefaultSysSpec = SysSpec{
 	Cgroups: []string{},
 	RuntimeSpec: RuntimeSpec{
 		DockerSpec: &DockerSpec{
-			Version:     []string{`18\.06\..*`}, //Requires [18.06] or later
+			Version:     []string{`18\.0[6,9]\..*`}, //Requires [18.09] or later
```
Comment (neolit123) [inline cmd/kubeadm/app/util/system/types_windows.go]: let's remove this comment too, i guess.
because we now support two versions.

Comment (xichengliudui) [reply cmd/kubeadm/app/util/system/types_windows.go]: I understand what you mean. Thank you for your quick reply

**Comment by xichengliudui** [comment]
@neolit123  Updated, you are really awesome in k8s!

File: cmd/kubeadm/app/util/system/types_unix.go [hunk 5]
```diff
@@ -62,7 +62,7 @@ var DefaultSysSpec = SysSpec{
 	Cgroups: []string{"cpu", "cpuacct", "cpuset", "devices", "freezer", "memory"},
 	RuntimeSpec: RuntimeSpec{
 		DockerSpec: &DockerSpec{
-			Version:     []string{`1\.1[1-3]\..*`, `17\.0[3,6,9]\..*`, `18\.06\..*`},
+			Version:     []string{`1\.1[1-3]\..*`, `17\.0[3,6,9]\..*`, `18\.0[6,9]\..*`},
```
Comment (neolit123) [inline cmd/kubeadm/app/util/system/types_unix.go]: there seems to be an extra space after the last `,` on this line.
the gofmt check might fail.

you can try running:
`./hack/update-gofmt.sh`

**Comment by neolit123** [comment]
thanks @xichengliudui 
sorry for the multiple requests for changes.

/lgtm
/assign @timothysc

**Comment by neolit123** [comment]
@kubernetes/sig-cluster-lifecycle-pr-reviews

**Comment by neolit123** [comment]
please run `./hack/update-gofmt.sh`

**Comment by xichengliudui** [comment]
@neolit123  I executed gofmt, thank you

**Comment by xichengliudui** [comment]
t p 6uvz9pc d d czd6

**Comment by xichengliudui** [comment]
I am a little confused now, I have not encountered such a situation before

**Comment by xichengliudui** [comment]
How should this command be used?

**Comment by neolit123** [comment]
@xichengliudui 
hello, the command should be called in the kubernetes source folder:
https://github.com/kubernetes/kubernetes/blob/master/hack/update-gofmt.sh

but your PR seems to pass CI now.
it is waiting on approval at this point.
we will discuss this PR later today in a meeting and possibly merge it!

thank you!
/retest

**Comment by neolit123** [comment]
actually, given this PR is technically just a fix (e.g. unit tests) for the PR by dims, i'm going to let it go.
at the same time i will notify the SIG that we are using 18.09 as recommended now.
/approve

**Comment by neolit123** [comment]
as long as `pull-kubernetes-verify` passes the PR is good.
the rest can flake (fail randomly).
/retest

**Pull request merged**